haystack/tracing/datadog.py

```diff
@@ -43,20 +43,9 @@ def raw_span(self) -> Any:
 
     def get_correlation_data_for_logs(self) -> dict[str, Any]:
         """Return a dictionary with correlation data for logs."""
-        raw_span = self.raw_span()
-        if not raw_span:
-            return {}
 
         # https://docs.datadoghq.com/tracing/other_telemetry/connect_logs_and_traces/python/#no-standard-library-logging
-        trace_id, span_id = (str((1 << 64) - 1 & raw_span.trace_id), raw_span.span_id)
-
-        return {
-            "dd.trace_id": trace_id,
-            "dd.span_id": span_id,
-            "dd.service": ddtrace.config.service or "",
-            "dd.env": ddtrace.config.env or "",
-            "dd.version": ddtrace.config.version or "",
-        }
+        return ddtrace.tracer.get_log_correlation_context()
```

Comment:
this method also takes care of returning a default if there's no active span

Reply:
No, but it might be a good idea to also involve someone from platform. (I'm not an expert)
I would like to merge a fix to stop tests failures, but we can later ask someone. WDYT?